rust/arrow/src/array.rs

```diff
@@ -370,6 +524,94 @@ def_numeric_from_vec!(UInt32Type, u32, DataType::UInt32);
 def_numeric_from_vec!(UInt64Type, u64, DataType::UInt64);
 def_numeric_from_vec!(Float32Type, f32, DataType::Float32);
 def_numeric_from_vec!(Float64Type, f64, DataType::Float64);
+// TODO: add temporal arrays
+
+macro_rules! def_temporal_from_vec {
```

Comment:
can we reuse `def_numeric_from_vec`? we just need to change `path` to `expr`. The macro name may also need to be changed.

Reply:
I didn't know that with `expr` and `path`, I would have reused the macros from the onset. Thanks!